bound/kt/src/main/kotlin/web5/sdk/crypto/verifiers/Verifier.kt

```diff
@@ -1,5 +1,30 @@
 package web5.sdk.crypto.verifiers
 
 import web5.sdk.rust.Verifier as RustCoreVerifier
+import web5.sdk.rust.Web5Exception
 
-typealias Verifier = RustCoreVerifier
\ No newline at end of file
+/**
+ * Set of functionality required to implement to be a compatible DSA verifier.
+ */
+interface Verifier {
+    /**
+     * Execute the verification of the signature against the payload by using the encapsulated public key material.
+     *
+     * @param message the data which was signed over.
+     * @param signature the signature over the message.
+     * @throws Web5Exception in the case of a failed verification.
+     */
+    fun verify(message: ByteArray, signature: ByteArray)
+}
+
+internal class ToOuterVerifier(private val rustCoreVerifier: RustCoreVerifier) : Verifier {
+    override fun verify(message: ByteArray, signature: ByteArray) {
+        rustCoreVerifier.verify(message, signature)
+    }
+}
+
+internal class ToInnerVerifier(private val verifier: Verifier) : RustCoreVerifier {
+    override fun verify(message: ByteArray, signature: ByteArray) {
+        verifier.verify(message, signature)
+    }
+}
```

Comment:
Having trouble understanding the intended use for these based on the name. `ToYadayada` sounds like a function name. If the intention is to convert from `Verifier` to `RustCoreVerifier`, why use a class instead of a function?

Reply:
Yup it's a great question! The overarching requirement is to enable developers to build-their-own `Verifier`, in this case, but it also applies to other interfaces such as `KeyManager` or `Signer`. Basically, these two concrete (and `internal` -- that's important) implementations are translation layers to-and-from the UniFFI generated interfaces. We don't want the consuming apps to be directly utilizing the UniFFI code gen, so we have to wrap around it, and to wrap an interface we have to concretize it with a simple wrapper implementation.